src/subscription_manager/gui/registergui.py

```diff
@@ -277,17 +324,16 @@ def __init__(self, backend, facts, reg_info=None,
         # To update the 'next/register' button in the parent dialog based on the new page
         self.register_notebook.connect('switch-page',
                                        self._on_switch_page)
-
-        screen_classes = [ChooseServerScreen, ActivationKeyScreen,
-                          CredentialsScreen, OrganizationScreen,
-                          EnvironmentScreen, PerformRegisterScreen,
-                          SelectSLAScreen, ConfirmSubscriptionsScreen,
-                          PerformSubscribeScreen, RefreshSubscriptionsScreen,
-                          InfoScreen, DoneScreen]
+        self.screen_classes = screen_classes or [ChooseServerScreen, ActivationKeyScreen,
```

Comment:
I think my preference would to be to include PerformUnregisterScreen in the screens for RegisterWidget, and just let a flag in RegisterInfo override it. 

For uxd purposes, I suspect it may make sense to add two screens, a 'ConfirmUnregisterScreen' gui screen, and a PerformUnregisterScreen (non-gui). 

They would each get visited by default, but would change behavior based on a RegisterInfo property ('confirm-register'?).  Then the parent dialog/app gets to define the behavior. It would also allow initial-setup/anaconda to easily add a 'unregister' step.